src/string/string-hashing.md

```diff
@@ -188,7 +188,7 @@ Implementation:
             long long cur_h = h [i + l-1];
             if (i) cur_h - = h [i-1];
             // Get all the hashes to the same degree
-            cur_h * = p_pow [ni-1];
```

Comment:
Okay .  Yes but in the current article there is ni-1 which should be n-i-1.  A small correction.